test/e2e/config_test.go

```diff
@@ -191,15 +194,20 @@ secrets:
 			RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
 
 		secrets := NewPresentClusterResource("serviceaccount", "test-sa-with-secrets", env.Namespace, kubectl).RawPath(ctlres.NewPathFromStrings([]string{"secrets"})).([]interface{})
-		require.Len(t, secrets, 2, "Expected one set and one generated secret")
+		if minorVersion < 24 {
```

Comment:
I didn't like adding so many conditions for the minorVersion, I will see if I can find any other alternative to it.